### PR TITLE
Trim docs in events specification

### PR DESCRIPTION
**Pull request by TomaszWaszczyk**
This PR closes: https://github.com/paritytech/ink/issues/1448

PS. I am not able directly to assign as a reviewer @SkymanOne.

### CODE REVIEW

**Review by SkymanOne — APPROVED** [review]

**Review by ascjones — COMMENTED** [review]
Please add a simple test for this, should be able to base it on one of the existing ones e.g the one from #1385 

https://github.com/paritytech/ink/blob/7ddf87f4294d09cbb023ce2df35e40376a912e18/crates/metadata/src/tests.rs#L213

**Comment by SkymanOne** [comment]
I see that CI fails. Can you please merge master head into your branch?

File: crates/metadata/src/tests.rs [hunk 1]
```diff
@@ -238,7 +238,7 @@ fn trim_docs_with_code() {
             "selector": "0x075bcd15",
             "args": [],
             "docs": [
-                "Example",
+                " Example",
```
Comment (SkymanOne) [inline crates/metadata/src/tests.rs:241]: Why add a whitespace here?

**Review by SkymanOne — CHANGES_REQUESTED** [review]
Looks good so far, please merge master into your branch and make sure CI passes :)

File: crates/metadata/src/tests.rs [hunk 2]
```diff
@@ -389,3 +389,35 @@ fn construct_runtime_contract_spec() {
     );
     assert_eq!(event_spec, expected_event_spec);
 }
+
+/// Test for trimming in docs for event specification
+#[test]
+fn trim_events_in_specification() {
```
Comment (ascjones) [inline crates/metadata/src/tests.rs:395]: This doesn't actually test the event docs, it is testing `ConstructorSpecBuilder::docs`

File: crates/metadata/src/tests.rs [hunk 3]
```diff
@@ -389,3 +389,35 @@ fn construct_runtime_contract_spec() {
...
+        .selector(123_456u32.to_be_bytes())
+        .docs(vec![" Foo event"])
+        .payable(Default::default())
+        .done();
+    let mut registry = Registry::new();
+    let compact_spec = constructor_spec.into_portable(&mut registry);
+
+    // WHEN
+    let json = serde_json::to_value(&compact_spec).unwrap();
+    let deserialized: ConstructorSpec<PortableForm> =
```
Comment (ascjones) [inline crates/metadata/src/tests.rs:408]: Should be testing the `EventSpecBuilder` and `EventParamSpecBuilder` which you have changed.

**Comment by HCastano** [comment]
@TomaszWaszczyk or @SkymanOne do either one of you want to finish this up, or should we close this?

**Comment by TomaszWaszczyk** [comment]
I will back here when ready.

**Pull request closed without merging**